### ISSUE TITLE
const field inside class is not handled correctly on Windows

### ISSUE THREAD

**Issue by upsuper** [servo/rust-bindgen#271]
Test code:
```c++
class A {
  const size_t count;
};
```

On Windows, it generates:
```rust
#[repr(C)]
#[derive(Debug, Copy)]
pub struct A {
    pub count: A_size_t,
}
```
where `A_size_t` is never defined.

**Comment by emilio** [comment]
I don't think this would reproduce if `size_t` was defined, does it?

i.e., I can't reproduce if I instead:

```cpp
typedef unsigned size_t; // Or define it in some other way.

class A {
  const size_t count;
};
```

**Comment by upsuper** [comment]
It doesn't matter.

```c++
typedef unsigned long long size_t;
class A {
  const size_t count;
};
```
still gives me
```rust
#[repr(C)]
#[derive(Debug, Copy)]
pub struct A {
    pub count: A_size_t,
}
```

FWIW, I can only reproduce this on Windows. It works fine on my macOS. I have no idea why...

**Comment by emilio** [comment]
Gah, github-debugging time again :).

Can you use `--emit-clang-ast` and paste it here?

**Comment by emilio** [comment]
The bindgen logs will also help, I guess.

**Comment by upsuper** [comment]
[**This**](https://github.com/servo/rust-bindgen/files/594548/log.txt) is the log with `RUST_LOG=bindgen cargo run -- test.h --emit-clang-ast -- -x c++ -std=c++14`, and `test.h` is the file I pasted in my previous comment.

**Comment by emilio** [comment]
Can you use `RUST_LOG=libbindgen` (the crate was split in #204)? Also, it seems clang is ignoring the `-x c++` argument, per the first warning, does it reproduce if instead of `test.h` it you call the file `test.hpp`?

**Comment by upsuper** [comment]
This is [log.txt](https://github.com/servo/rust-bindgen/files/594643/log.txt) from `RUST_LOG=bindgen,libbindgen cargo run -- test.hpp --emit-clang-ast -- -std=c++14`.

**Comment by emilio** [comment]
So it's again an MSVC backend bug, where they report no USR for the typedef declaration, so we don't find it and assume that "const size_t" is a type declared inside of `A`...

**Comment by emilio** [comment]
@upsuper, does the following patch make it work for you?

```diff
diff --git a/libbindgen/src/ir/context.rs b/libbindgen/src/ir/context.rs
index a1ec8c7..bda3e3c 100644
--- a/libbindgen/src/ir/context.rs
+++ b/libbindgen/src/ir/context.rs
@@ -222,7 +222,7 @@ impl<'ctx> BindgenContext<'ctx> {
                 error!("Valid declaration with no USR: {:?}, {:?}",
                        declaration,
                        location);
-                return;
+                TypeKey::Declaration(declaration)
             };
 
             let old = self.types.insert(key, id);
```

**Comment by upsuper** [comment]
Yes, it seems to work. Thanks.

**Comment by emilio** [comment]
I added the fix in #260, should be merged soon.

Thank you for reporting it :)
